safe-modules/timelock-guard.md

```diff
@@ -0,0 +1,192 @@
+
+The `cancellation_threshold` should either always be the same as the `blocking_minority` or if the
+code complexity is acceptable, the `cancellation_threshold` can be dynamic between 1 and
+`blocking_minority`. A dynamic `cancellation_threshold` offers a better response time while not
+being vulnerable to sustained abuse by a small amount of malicious owners.
+
+If a dynamic `cancellation_threshold` is chosen, it would start at 1 and increase by 1 with each
+successful consecutive cancellation. The `cancellation_threshold` would reset to 1 with each
+successful execution of an scheduled transaction.
+
+## Alternatives Considered
+<!-- Describe any alternatives that were considered during the development of this design. Explain
+why the alternative designs were ultimately not chosen and where they failed to meet the product
+requirements. -->
+
+Other potential designs here make trade-offs for security or usability that are undesireable enough
+to not really be worth considering. Examples of this are:
+
+- Allowing any user to reveal a transaction for later execution, which creates more considerations
+  for monitoring/offchain infrastructure and changes the transaction flow.
+- Having multisigs queue transaction via the timelock directly and then allowing anyone to execute
+  the transaction via a module, which changes the signing flow significantly and means transactions
+  are challenging to simulate properly.
+- Configurable `cancellation_thresholds` were considered, but the risk of misconfiguration over
+  time was expected to be considerable.
+- A fixed `cancellation_threshold` of 1 was considered, but the risk of operational disruption by
+  malicious actors was considered unacceptable.
+- An increasing `cancellation_threshold` from 1 to `min(blocking_minority, quorum)`, which would
+  increase by one with each successful cancellation, and reset to one with each successful
+  execution, was considered. It was deemed that at its lower end it would be too operationally
+  disruptive, and not worth the additional complexity.
+
+We could not find any alternative design with the simplicity and security of the one that was
+ultimately presented in this document.
+
+## Risks and Uncertainties
+
+<!-- Explain any risks and uncertainties that this design includes. Highlight aspects of the design
+that remain unclear and any potential issues we may face down the line. -->
+
+### Timelock Overhead
+Any type of mandatory timelock introduces overhead for multisigs that want to execute transactions
+in a timely fashion. Multisig leads will have to plan ahead of time in a way that they were not
+previously required to do. We can mitigate some of this by allowing the full threshold to execute
+faster, but this will be difficult to coordinate regularly and leads should not expect this to be
+a norm.
+
+### Time Critical Transactions
+Some multisigs may face scenarios where they need to be able to execute transactions very quickly
```

Comment:
one scenario for which ive seen a need for time critical transactions is when a mistake is made and some kind of timely surgical recovery is possible. are we making the assumption that the new review period via TimelockGuard delay implies no mistakes? or are we assuming that they can be acceptably managed via Pause?

Reply:
Bingo, we realized this as well in our threat modelling yesterday. We might have to rethink the feature.

The Pause does not cover all possible safety incidents, so we need an emergency response path.